server/recording_api.go

```diff
@@ -36,6 +36,11 @@ func (p *Plugin) recJobTimeoutChecker(callID, jobID string) {
 	// If the recording hasn't started (bot hasn't joined yet) we notify the
 	// client.
 	if recState.StartAt == 0 {
+		if recState.JobID != jobID {
+			p.LogInfo("a new job has started in between, exiting", "callID", callID, "jobID", jobID)
+			return
+		}
```

Comment:
Adding an extra check to avoid making potentially inconsistent state updates due to previously failed jobs. Ideally we'd cancel this timeout check whenever we stop a recording but the synchronization needed would make the code more complex so it seems like a few extra goroutines that return early are not too bad.